File: csharp/DiagramPrinter/DiagramMetadata.cs
namespace DiagramPrinter;

public class DiagramMetadata
{
    public string FullFilename { get; protected set; }
    public string FileType { get; protected set; }
    public bool FileAvailable { get; protected set; }

    public DiagramMetadata(FlowchartDiagram diagram)
    {
        FullFilename = diagram.Name() + "_" + diagram.SerialNumber();
        FileType = diagram.Name().Contains("Flowchart") ? "PDF" : "Spreadsheet";
        FileAvailable = !diagram.IsDisposed();
    }

    protected DiagramMetadata()
    {
        // for subclass for testing
    }
}

File: java/src/main/java/sammancoaching/DiagramMetadata.java
package sammancoaching;

public class DiagramMetadata {
    protected String fullFilename;
    protected String fileType;
    protected boolean fileAvailable;

    public DiagramMetadata(FlowchartDiagram diagram) {
        this.fullFilename = diagram.getName() + "_" + diagram.getSerialNumber();
        this.fileType = diagram.getName().contains("Flowchart") ? "PDF" : "Spreadsheet";
        this.fileAvailable = !diagram.isDisposed();
    }

    protected DiagramMetadata() {
        // Enable subclassing for testing purposes
    }
}
